### PR TITLE
pcr: add crdb_internal.cluster_replication_spans view

### PR DESCRIPTION
**Pull request by dt**
Release note: none.
Epic: none.

### CODE REVIEW

File: pkg/sql/crdb_internal.go [hunk 1]
```diff
@@ -8929,3 +8930,26 @@ func populateFlowTokensResponse(
...
 	return nil
 }
+
+var crdbInternalClusterReplicationResolvedView = virtualSchemaView{
+	schema: `
+		CREATE VIEW crdb_internal.cluster_replication_spans AS WITH spans AS (
+			SELECT
+				job_id, jsonb_array_elements(crdb_internal.pb_to_json('progress', value)->'streamIngest'->'checkpoint'->'resolvedSpans') AS s
+			FROM system.job_info
+			WHERE info_key = 'legacy_progress' AND crdb_internal.pb_to_json('progress', value)->'streamIngest' IS NOT NULL
```
Comment (stevendanna) [inline pkg/sql/crdb_internal.go:8940]: Interesting. I suppose on the one hand it is nice for this query to only touch the job table. On the other hand I have to imagine that in some cases something like `WHERE job_id IN (SELECT id FROM system.jobs WHERE type = 'STREAM INGESTION'`) (or whatever the right type is) might be faster than than scanning every progress.

I leave that to your judgement. We can always change it later.

**Review by stevendanna — APPROVED** [review]
Overall I like this. I do wonder if the same should have `physical_replication` in it as to avoid confusion. But I don't feel strongly.

**Comment by dt** [comment]
> physical_replication

I think "cluster" in "cluster replication" is sufficient? I think unqualified "replication" is and always should be assumed to be *kv* raft replication, but I think "cluster replication" is clean enough. It's an internal debug tool so I'm inclined to optimize a little more for fitting the things I want on screen over being verbose for wider audiences?

**Comment by dt** [comment]
TFTR!

bors r+

**Pull request merged**